05-how-to-contribute-to-open-source.md

```diff
@@ -25,161 +29,321 @@ OpenSauced is a powerful tool for finding open source projects to contribute to.
 
-3. **Start small**: When you're new to a project, it's a good idea to start with small, manageable tasks, such as fixing bugs, adding tests, or updating documentation. This will help you become familiar with the codebase and development workflow without getting overwhelmed.
+- **A screenshot or screen recording when you make a UI change**: Providing screenshots or screen recordings will make it easier for maintenance to visualize your changes and review your pull request.
 
-4. **Ask for help**: If you're unsure about something or encounter a problem, don't hesitate to ask for help. Open source communities are generally supportive and welcoming, and other contributors will be happy to assist you.
+#### 3. Resolving Merge Conflicts Immediately
 
-5. **Be patient and persistent**: Onboarding in a new project can be challenging, especially if you're new to open source development. Be patient with yourself, and don't be discouraged by setbacks or mistakes. With persistence and practice, you'll become more comfortable and confident in your contributions.
+If a branch has merge conflicts that must be resolved, the merge button is automatically disabled. So, maintainers are not able to merge the pull request.
+
+When you notice merge conflicts in your pull request or if a maintainer asks you to resolve merge conflicts, fix them immediately. The sooner you resolve the conflicts, the sooner maintainers can review and merge your pull request.
+
+### Merge Conflicts in the Guest Book Repository
+
+Since the primary purpose of the guest book is to add your name to `.all-contributorsrc` and the `README.md` files, there is a high chance that you will encounter merge conflicts.
+
+The conflicts happen when maintainers have merged pull requests before yours while you're working on your changes or waiting for your pull request to be reviewed. And you need to resolve them before your pull request can be merged.
+
+#### Resolving Merge Conflicts
+
+Before resolving merge conflicts, you must first [update your branches](#updating-branches). Then, follow these steps:
+
+1. In the `.all-contributorsrc` file:
+
+   - Click the "Accept Both Change" option on the top of your workspace in VS Code.
+   - Move your profile details to the end of the contributors' array and fix anything necessary.
+
+2. In the `README.md` file:
+
+   - Click the "Accept Incoming Change" option on the top of your workspace in VS Code for each conflict in this file.
+
+3. Run `npm run contributors:generate`.
+
+   You will now see that the all contributors badge has been incremented, and your profile is generated at the end of the contributors' list in the `README.md` file.
+
+4. Add and commit your changes.
+
+   ```bash
+   git commit -am "Resolve merge conflicts"
+   ```
+
+5. Push your commits to your remote branch.
+
+   ```bash
+   git push
+   ```
 
 ## What Happens Next?
```

Comment:
For better flow, I'm thinking to move this section to before "Let's Get Practical". So from the "Let's Get Practical" forward will contain all technical sections. Any thoughts?

Reply:
I agree.

Reply:
I second this! :)